hdfs/src/test/scala/akka/stream/alpakka/hdfs/HdfsWriterSpec.scala

```diff
@@ -0,0 +1,501 @@
+      files.map(_.getLen).sum shouldEqual 1024
+      files.foreach(f => f.getLen should be <= 512L)
+
+      verifyOutputFileSize(fs, logs)
+      readLogsWithFlatten(fs, logs) shouldBe data.flatMap(_.utf8String)
+    }
+
+    "detect upstream finish and move remaining data to hdfs" in {
+      // Use huge rotation
+      val data = generateFakeContent(1, FileUnit.KB.byteCount)
+      val dataIterator = data.toIterator
+      //#define-data
+      val flow = HdfsFlow.data(
+        fs,
+        SyncStrategy.count(500),
+        RotationStrategy.size(1, FileUnit.GB),
+        HdfsWritingSettings()
+      )
+      //#define-data
+      val resF = Source
+        .fromIterator(() => dataIterator)
+        .map(IncomingMessage(_))
+        .via(flow)
+        .runWith(Sink.seq)
+
+      val logs = Await.result(resF, Duration.Inf)
+      logs.size shouldEqual 1
+
+      getFiles(fs).head.getLen shouldEqual 1024
+
+      verifyOutputFileSize(fs, logs)
+      readLogsWithFlatten(fs, logs) shouldBe data.flatMap(_.utf8String)
+    }
+
+    "use buffer rotation and produce three files" in {
+      val flow = HdfsFlow.data(
+        fs,
+        SyncStrategy.count(1),
+        RotationStrategy.count(2),
+        settings
+      )
+
+      val resF = Source
+        .fromIterator(() => books.toIterator)
+        .map(IncomingMessage(_))
+        .via(flow)
+        .runWith(Sink.seq)
+
+      val logs = Await.result(resF, Duration.Inf)
+      logs.size shouldEqual 3
+
+      verifyOutputFileSize(fs, logs)
+      readLogs(fs, logs) shouldBe books.map(_.utf8String).grouped(2).map(_.mkString).toSeq
+    }
+
+    "use time rotation" in {
+      val (cancellable, resF) = Source
+        .tick(0.millis, 50.milliseconds, ByteString("I love Alpakka!"))
+        .map(IncomingMessage(_))
+        .via(
+          HdfsFlow.data(
+            fs,
+            SyncStrategy.none,
+            RotationStrategy.time(500.milliseconds),
+            HdfsWritingSettings()
+          )
+        )
+        .toMat(Sink.seq)(Keep.both)
+        .run()
+
+      system.scheduler.scheduleOnce(1500.milliseconds)(cancellable.cancel()) // cancel within 1500 milliseconds
+      val logs = Await.result(resF, Duration.Inf)
+      verifyOutputFileSize(fs, logs)
+      List(3, 4) should contain(logs.size)
+    }
+
+    "should use no rotation and produce one file" in {
+      val flow = HdfsFlow.data(
+        fs,
+        SyncStrategy.none,
+        RotationStrategy.none,
+        HdfsWritingSettings()
+      )
+
+      val resF = Source
+        .fromIterator(() => books.toIterator)
+        .map(IncomingMessage(_))
+        .via(flow)
+        .runWith(Sink.seq)
+
+      val logs = Await.result(resF, Duration.Inf)
+      logs.size shouldEqual 1
+
+      val files = getFiles(fs)
+      files.size shouldEqual logs.size
+      files.head.getLen shouldEqual books.map(_.toArray.length).sum
+    }
+
+    "kafka-example - store data with passThrough" in {
+      //#define-kafka-classes
```

Comment:
Move the documentation examples to a package like `docs.scaladsl` and `docs.javadsl` to make sure everything you show is accessible.

Reply:
Actually, this is not a real example, and it contains test assertions for pass-through logic. I would like to have it as part of tests. Shall we keep it as it is?